### ISSUE TITLE
decodeB64 does not work in Android

### ISSUE THREAD

**Issue by yuyaogawa** [deeplearning4j/deeplearning4j#3603]
#### Issue Description

decodeB64 does not work in Android because of "org.apache.commons.codec.binary.Base64"

#### Version Information

Please indicate relevant versions, including, if relevant:

* Deeplearning4j version
* platform information (OS, etc)
* CUDA version, if used
* NVIDIA driver version, if in use

#### Contributing

Replace "org.apache.commons.codec.binary.Base64" for "android.util.Base64"

public static String decodeB64(String word) {
        if (word.startsWith("B64:")) {
            String arp = word.replaceFirst("B64:","");
            try {
                return new String(Base64.decode(arp, 0), "UTF-8"); <<<<< changed
            } catch (Exception e) {
                throw new RuntimeException(e);
            }
        } else return word;
    }

**Comment by agibsonccc** [comment]
You're asking us to make an android specific change..there has to be something better we could do here. I can't justify that. Especially when the bulk of the usage for word2vec is on servers not embedded.

How big is your word2vec model here? I agree it should run on there but your suggestion isn't the right way to go here..I would be ok with possible abstracting it in some way though.

@saudet what do you think?

**Comment by raver119** [comment]
Let's make something abstract here, but I'm afraid w2v on android will have other issues as well...

**Comment by agibsonccc** [comment]
@yuyaogawa you didn't need to close it :D. We just need a better way of doing it than what you're suggesting here. We can't mainline an android specific change. Could you tell us more about your use case? Usually word2vec models are too big for phones..

**Comment by yuyaogawa** [comment]
Ok, so I am new here to use Github and I don't know how to cooperate...

Now, I am working on a news app for android that analyzes news content whether it makes people smile or not. I know it is good way to implement deep learning systems on a server but I want to implement it on android.

**Comment by agibsonccc** [comment]
@yuyaogawa yes of course no problem! That is what we are here for. How big is your word2vec model though? 

Depending on how big the model it it could be as large as a few gigs. Have you measured any of this just on a desktop at all?

**Comment by yuyaogawa** [comment]
Yeah it is about 300MB

**Comment by yuyaogawa** [comment]
@agibsonccc how do you make an android app that use deep learning process?
I just want to use the doc2vec parameter that has already learning.

**Comment by raver119** [comment]
I'd still suggest to handle vectorization on server, and serve it via rest/graphql/whatever. Just keep vocab on phone.

**Comment by agibsonccc** [comment]
@yuyaogawa 300 MB isn't terrible but that's still a ton of ram for a phone app. I would follow raver's suggestion and build a rest app. In the mean time we will implement a version of your suggestion.

Thanks for raising an issue!

**Comment by saudet** [comment]
Abstraction sounds like the way to go to me!

**Comment by agibsonccc** [comment]
Closing — this issue is over 7 years old with no recent activity. The codebase has undergone a major rewrite. If this is still relevant, please open a new issue with current reproduction steps.
